cmd/runtimetest/main.go

```diff
@@ -370,6 +371,52 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func getActualCgroupPath() (string, error) {
+	f, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return "", err
+		}
+
+		text := s.Text()
+		parts := strings.Split(text, ":")
+		return parts[2], nil
+	}
+	return "", fmt.Errorf("can not get cgroup path")
+}
+
+func validateCgroupsPath(spec *rspec.Spec) error {
+	logrus.Debugf("validating cgroupsPath")
+	expectedPath := spec.Linux.CgroupsPath
+	if expectedPath == nil {
+		return nil
+	}
+	*expectedPath = strings.Replace(*expectedPath, ":", "/", -1)
+
+	actualPath, err := getActualCgroupPath()
+	if err != nil {
+		return err
+	}
+
+	if filepath.IsAbs(*expectedPath) {
+		if *expectedPath != actualPath {
+			return fmt.Errorf("Cgroup path expected: %v, actual: %v", *expectedPath, actualPath)
+		}
+	} else {
+		*expectedPath = strings.Replace(*expectedPath, "../", "", -1)
```

Comment:
On Mon, Aug 01, 2016 at 01:24:04AM -0700, Aleksa Sarai wrote:
> > +	if filepath.IsAbs(*expectedPath) {
> > +		if *expectedPath != actualPath {
> > +			return fmt.Errorf("Cgroup path expected: %v, actual: %v", *expectedPath, actualPath)
> > +		}
> > +	} else {
> > +		*expectedPath = strings.Replace(*expectedPath, "../", "", -1)
> 
> In fact, in runC we clean the `cgroupsPath` quite strongly to avoid
> path traversal vulnerabilities. So I have the feeling that we would
> fail this test case from the runC side.

The not-absolute spec is an unfailable MAY, so no worries ;).